services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -86,10 +87,17 @@ public SubmissionPayloadValidator(SubmissionServiceConfig submissionServiceConfi
     public boolean isValid(SubmissionPayload submissionPayload, ConstraintValidatorContext validatorContext) {
       List<TemporaryExposureKey> exposureKeys = submissionPayload.getKeysList();
       validatorContext.disableDefaultConstraintViolation();
-
-      boolean isValid = checkKeyCollectionSize(exposureKeys, validatorContext);
-      isValid &= checkUniqueStartIntervalNumbers(exposureKeys, validatorContext);
-      isValid &= checkNoOverlapsInTimeWindow(exposureKeys, validatorContext);
+      boolean isValid = false;
+
+      for (TemporaryExposureKey exposureKey: exposureKeys) {
```

Comment:
Something doesn't look right here. For each exposure key in the payload we check the constraints of the entire set? I'm referring to the else branch. Can you confirm that is intended?